### PR TITLE
feat: add agentless to generate command for AWS

### PR DESCRIPTION
**Pull request by PengyuanZhao**
## Summary

- Add the agentless support to the `lacework generate cloud-account aws` command
- This PR adds the support for agentless [single-account-multi-region](https://github.com/lacework/terraform-aws-agentless-scanning/blob/main/examples/single-account-multi-region/main.tf)
- Will handle [multi-account-multi-region](https://github.com/lacework/terraform-aws-agentless-scanning/tree/main/examples/multi-account-multi-region) in following PRs.

## How did you test this change?
`make test`

`lacework generate cloud-account aws --agentless --config --cloudtrail`:

https://github.com/lacework/go-sdk/assets/7945666/4cf7c4ea-174d-47b1-8c7b-8c029dc54083

Generated terraform code:

```
terraform {
  required_providers {
    lacework = {
      source  = "lacework/lacework"
      version = "~> 1.0"
    }
  }
}

provider "aws" {
  alias   = "main"
  profile = "main"
  region  = "us-east-1"
}

provider "aws" {
  alias   = "subaccount1"
  profile = "subaccount1"
  region  = "us-east-2"
}

provider "aws" {
  alias   = "subaccount2"
  profile = "subaccount2"
  region  = "us-east-3"
}

module "aws_config" {
  source  = "lacework/config/aws"
  version = "~> 0.5"

  providers = {
    aws = aws.main
  }
}

module "aws_config_subaccount1" {
  source  = "lacework/config/aws"
  version = "~> 0.5"

  providers = {
    aws = aws.subaccount1
  }
}

module "aws_config_subaccount2" {
  source  = "lacework/config/aws"
  version = "~> 0.5"

  providers = {
    aws = aws.subaccount2
  }
}

module "main_cloudtrail" {
  source                = "lacework/cloudtrail/aws"
  version               = "~> 2.7"
  iam_role_arn          = module.aws_config.iam_role_arn
  iam_role_external_id  = module.aws_config.external_id
  iam_role_name         = module.aws_config.iam_role_name
  use_existing_iam_role = true

  providers = {
    aws = aws.main
  }
}

module "lacework_aws_agentless_scanning_global" {
  source                    = "lacework/agentless-scanning/aws"
  version                   = "~> 0.6"
  global                    = true
  regional.               = true
  lacework_integration_name = "my_agentless_integration"
}

module "lacework_aws_agentless_scanning_region_subaccount1" {
  source                  = "lacework/agentless-scanning/aws"
  version                 = "~> 0.6"
  global_module_reference = module.lacework_aws_agentless_scanning_global
  regional                = true

  providers = {
    aws = aws.subaccount1
  }
}

module "lacework_aws_agentless_scanning_region_subaccount2" {
  source                  = "lacework/agentless-scanning/aws"
  version                 = "~> 0.6"
  global_module_reference = module.lacework_aws_agentless_scanning_global
  regional                = true

  providers = {
    aws = aws.subaccount2
  }
}
```

### CODE REVIEW

**Review by jon-stewart — APPROVED** [review]

File: cli/cmd/generate_aws.go [hunk 1]
```diff
@@ -770,6 +792,11 @@ func askAdvancedAwsOptions(config *aws.GenerateAwsTfConfigurationArgs, extraStat
 		// we can have other accounts even if we only have Config integration (Scenario 7)
 		var options []string
 
+		// Only show Advanced CloudTrail options if CloudTrail integration is set to true
```
Comment (afiune) [inline cli/cmd/generate_aws.go:795]:
```suggestion
		// Only show Advanced Agentless options if Agentless integration is set to true
```

File: cli/cmd/generate_aws.go [hunk 2]
```diff
@@ -15,6 +15,8 @@ import (
 
 var (
 	// Define question text here so they can be reused in testing
+	QuestionEnableAgentless             = "Enable agentless integration?"
```
Comment (afiune) [inline cli/cmd/generate_aws.go:18]:
```suggestion
	QuestionEnableAgentless             = "Enable Agentless integration?"
```

File: lwgenerate/aws/aws.go [hunk 3]
```diff
@@ -612,3 +636,60 @@ func createCloudtrail(args *GenerateAwsTfConfigurationArgs) (*hclwrite.Block, er
...
 }
+
+func createAgentless(args *GenerateAwsTfConfigurationArgs) ([]*hclwrite.Block, error) {
+	if !args.Agentless {
+		return nil, nil
+	}
+
+	blocks := []*hclwrite.Block{}
+
+	integrationName := "agentless_from_terraform"
```
Comment (afiune) [inline lwgenerate/aws/aws.go:647]: I don't think we need to pass the integration name if it is not specified by the user, at
least I see a default value at https://github.com/lacework/terraform-aws-agentless-scanning/blob/main/variables.tf#L4

If we want to always customize it, let's just call it something like this:
```suggestion
	integrationName := "agentless-cli-deployment"
```
that way we will know it is coming from the CLI deployment

Comment (PengyuanZhao) [reply lwgenerate/aws/aws.go]: Removed this to keep it simple.

**Review by afiune — APPROVED** [review]
Later, not now, we need to re-use the IAM role like this https://gist.github.com/afiune/08a4ef4547dab13ce2c1367a122543a6#file-aws-tf-L93-L96

For now, this is good enough, let's merge and release ASAP, please.

**Pull request merged**